noxfile.py

```diff
@@ -0,0 +1,261 @@
+    """Open the coverage report in the browser"""
+    session.install("coverage")
+    session.run("coverage", "html")
+    session.run("xdg-open", "htmlcov/index.html", external=True)
+
+
+def _run_integration_tests(session: Session, env: dict[str, str]) -> None:
+    _install_requirements(session)
+    default_env = {
+        "CACHI2_TEST_NETRC_CONTENT": "machine 127.0.0.1 login cachi2-user password cachi2-pass",
+        "CACHI2_TEST_LOCAL_PYPISERVER": "true",
+        "CACHI2_TEST_LOCAL_DNF_SERVER": "true",
+    }
+    default_env.update(env)
+    session.run(
+        "pytest",
+        "--log-cli-level=WARNING",
+        "tests/integration",
+        *session.posargs,
+        env=default_env,
+    )
+
+
+@nox.session(name="integration-tests", reuse_venv=True)
+def integration_tests(session: Session) -> None:
+    """Run integration tests only for the affected code base in the current branch"""
+    _run_integration_tests(
+        session,
+        {},
+    )
+
+
+@nox.session(name="all-integration-tests", reuse_venv=True)
+def all_integration_tests(session: Session) -> None:
+    """Run all integration tests that are available"""
+    _run_integration_tests(
+        session,
+        {
+            "CACHI2_RUN_ALL_INTEGRATION_TESTS": "true",
+        },
+    )
+
+
+@nox.session(name="generate-test-data", reuse_venv=True)
+def generate_test_data(session: Session) -> None:
+    """Run all integration tests that are available and update SBOMs"""
+    _run_integration_tests(
+        session,
+        {
+            "CACHI2_RUN_ALL_INTEGRATION_TESTS": "true",
+            "CACHI2_GENERATE_TEST_DATA": "true",
+        },
+    )
+
+
+@nox.session(name="build-image", reuse_venv=True)
+def build_image(session: Session) -> None:
+    """Build the cachi2 container image"""
+    session.run(
+        "podman",
+        "build",
+        "--tag",
+        "localhost/cachi2:latest",
+        ".",
+        external=True,
+    )
+
+
+@nox.session(name="build-pristine-image", reuse_venv=True)
+def build_pristine_image(session: Session) -> None:
+    """Build the cachi2 container image from scratch"""
+    session.run(
+        "podman",
+        "build",
+        "--pull-always",
+        "--no-cache",
+        "--tag",
+        "localhost/cachi2:latest",
+        ".",
+        external=True,
+    )
+
+
+@nox.session(name="pip-compile", reuse_venv=True, python="3.9")
+def pip_compile(session: Session) -> None:
+    """Generate requirements.txt and requirements-extras.txt locally"""
+    session.install("pip-tools")
+    session.run(
+        "pip-compile",
+        "--generate-hashes",
+        "--output-file=requirements.txt",
+        "--rebuild",
+        "pyproject.toml",
+        silent=True,
+    )
+    session.run(
+        "pip-compile",
+        "--all-extras",
+        "--generate-hashes",
+        "--output-file=requirements-extras.txt",
+        "--rebuild",
+        "pyproject.toml",
+        silent=True,
+    )
```

Comment:
This one becomes confusing with the other target (podman) in context. Since we have a hard dependency on pip-compile to always run with 3.9, risking that a developer unaware of this rule will try submitting requirements updates which are easy to slip in during the review process, so I'd say this one should be dropped and the target should always run in a container as it's been the case for a while via the Makefile.

Reply:
I added this one as _optional_ or alternative if someone does not want to run a container.
> @nox.session(name="pip-compile", reuse_venv=True, python="3.9")

python="3.9" ensures that the session will not run if python3.9 is not installed locally

Reply:
Seems pointless to me doing it this way when you already have a solution in place that enforces the correct Python version regardless of the one installed locally AND works unconditionally all the time.